modules/ROOT/pages/work-with-metadata.adoc

```diff
@@ -0,0 +1,60 @@
+= Work With Metadata in Studio
+
+Anypoint Studio 7.10 infers the metadata of your current flow so that, when you use a Flow Reference component to reference another flow, you can set the metadata from your current flow as custom metadata for your referenced flow. By automatically detecting the actual metadata of your existing flow, Studio enables you to propagate the metadata in your flow to the flow you are referencing as the expected metadata.
+
+By leveraging Studio to define the metadata you can:
+
+* Automatically propagate the metadata of your flow to a referenced flow.
+* Easily define the metadata of each referenced flow within your application.
+* Set custom metadata in a new flow without manually defining it.
+
+When you use Studio to scaffold your flows based on a RAML specification, Studio infers the metadata of your flow based on your RAML specification's types. However, note that Union types are not supported
+
+Studio converts your inferred metadata as custom metadata for the flow you are referencing.
+
+For example, assume a flow named `flow-1` that uses a Set Payload component to create a JSON payload:
+
+.Actual Metadata Sample
+[source,JSON,linenums]
+--
+{
+  "value": {
+    "ID": 1,
+    "departureDate": "2016/12/20",
+    "emptySeats": 200,
+    "plane": {
+      "type": "Boeing 747",
+      "totalSeats": 400
+    }
+  }
+}
+--
+
+When you add a Flow Reference component to a second flow, `flow-2`, Studio detects the current metadata as the actual metadata that the referenced flow will receive and automatically assigns it as custom metadata in `flow-2`.
```

Comment:
I don't quite understand what it is said here, is this the case when we have a flow ref in flow 1 and then that flow ref points to flow 2? 
Also, automatic propagation only happens where there is no custom metadata in the referenced flow

Reply:
Yes, that would be that case. I should probably add:

> and automatically assigns it as custom metadata in `flow-2` if the flow does not already have custom metadata declared for it.

Or something similar.